### ISSUE TITLE
No warning on mutable return object in function signature

### ISSUE THREAD

**Issue by jonaskello** [jonaskello/eslint-plugin-functional#64]
This does not give any warning:

```ts
function renderOutdatedTableStructure({
  dispatch,
  translator,
  state,
}: {
  dispatch: Dispatch<Action>;
  translator: Translator;
  state: ProductTableEditorState;
}): JSX.Element {
...
}
```

In this case I think the return type should have `readonly` modifiers enforced even if `ignoreLocal` is true (input/output of the function is not part of "local").

```ts
function renderOutdatedTableStructure({
  dispatch,
  translator,
  state,
}: {
  readonly dispatch: Dispatch<Action>;
  readonly translator: Translator;
  readonly state: ProductTableEditorState;
}): JSX.Element {
...
}
```

**Comment by RebeccaStevens** [comment]
I believe the cause of this issue is the same as with #75.

#78 should fix both issues.
